std/regex/package.d

```diff
@@ -457,6 +460,17 @@ template ctRegexImpl(alias pattern, string flags=[])
     test(re);
 }
 
+@safe unittest
+{
+    auto re = ctRegex!`foo`;
+    assert(matchFirst("foo", re));
+
+    // test reassignment
+    re = ctRegex!`bar`;
+    assert(matchFirst("bar", re));
+    assert(!matchFirst("bar", ctRegex!`foo`));
```

Comment:
This one currently fails; `ctRegex!"foo"` behaves as `ctRegex!"bar"` after the reassignment.